File: blockchain/block.py
class Block:
    def __init__(self, index, previous_hash, timestamp, vote, hash):
        self.__index = index
        self.__previous_hash = previous_hash
        self.__timestamp = timestamp
        self.__vote = vote
        self.__hash = hash

    
    @property
    def index(self):
        return self.__index
    @property
    def previous_hash(self):
        return self.__previous_hash
    @property
    def timestamp(self):
        return self.__timestamp
    @property
    def vote(self):
        return self.__vote
    @property
    def hash(self):
        return self.__hash
    
    def __repr__(self):
        return f"Block(index={self.index}, previous_hash='{self.previous_hash}', timestamp={self.timestamp}, data='{self.data}', hash='{self.hash}')"

File: backend/server/block.js
import crypto from 'crypto';

export class Block {
    // Campos privados 
    #index;
    #vote;
    #previousHash;
    #timestamp;
    #hash;

    constructor(index, votes, previousHash) {
        this.#index = index;
        this.#vote = votes; // Esto será un array de votos
        this.#previousHash = previousHash;
        this.#timestamp = Math.floor(Date.now() / 1000); // Timestamp UNIX en segundos
        this.#hash = this.#calculateHash();
    }

    // Getters 
    get index() { return this.#index; }
    get previousHash() { return this.#previousHash; }
    get timestamp() { return this.#timestamp; }
    get vote() { return this.#vote; }
    get hash() { return this.#hash; }

    // Llama al método privado para calcular y devolver el hash.
    getCalculatedHash() {
        return this.#calculateHash();
    }

    // Método privado para calcular el hash del bloque (reemplaza a __calculate_hash)
    #calculateHash() {
        // JSON.stringify es el 'json.dumps' de JS
        const dataString = this.#index.toString() +
                         this.#previousHash +
                         this.#timestamp.toString() +
                         JSON.stringify(this.#vote); // Serializa el array de votos

        // 'crypto.createHash' es el 'hashlib.sha256' de JS
        return crypto.createHash('sha256').update(dataString).digest('hex');
    }

    toString() {
        return `Block(index=${this.#index}, previousHash='${this.#previousHash}', hash='${this.#hash}')`;
    }
}